bot.py

```diff
@@ -28,8 +80,19 @@ async def on_message(self, message):
         """ Handler for receiving messages """
         if message.author == self.user:
             return
-        if message.content.startswith('!hello'):
-            await message.channel.send('Hello!')
+
+        handler, command_content, error_msg = self.get_handler_if_command(message.content)
+
+        # There was an error during command parsing.
+        if error_msg:
+            return await message.channel.send(error_msg)
```

Comment:
We can ignore failed commands, and use `logging.error(error_msg)` to log on the host

Reply:
error_msg is only intended for signaling to the user that the command was not recognized. But I agree in handlers we should use logging.error etc.